phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -245,6 +247,14 @@ private void deleteRowIfAgedEnough(byte[] indexRowKey, List<Cell> row, long ts,
             if ((EnvironmentEdgeManager.currentTimeMillis() - ts) > ageThreshold) {
                 Delete del = new Delete(indexRowKey, ts);
                 if (specific) {
+                    // Get all the cells of this row
+                    deleteRowScan.withStartRow(indexRowKey, true);
```

Comment:
hmm...if you knew for sure what the CFs were you could probbaly do addFamilyVersion in a loop instead, possibly saving yourself a Scan. You wouldn't have to worry about dynamic columns (can't be indexed), but you'd need a PTable for getColumnFamilies, and to know for sure it was up to date as of the timestamp you're deleting.

I'm going to approve as-is, just thinking out loud here.

Reply:
It is safer to retrieve the columns to find out their timestamps and delete markers for these timestamps, at least for now. We can think about your suggestion in future.